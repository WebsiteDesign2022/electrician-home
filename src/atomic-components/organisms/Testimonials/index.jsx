import React from 'react';

import TitleWithLineMotif from '../../molecules/TitleWithLineMotif';

import Testimonial from '../../molecules/Testimonial';

import './index.scss';

const Testimonials = ({ testimonials }) => (
    <div className="testimonials">

        <TitleWithLineMotif title="Testimonials" />
        
        <div className="row row-cols-1 row-cols-md-3 gx-4 py-5 list">
            { testimonials.map(({ description, name }, index) => (
                <Testimonial description={description} name={name} key={index} />)) 
            }
        </div>
    </div>
);

export default Testimonials;
