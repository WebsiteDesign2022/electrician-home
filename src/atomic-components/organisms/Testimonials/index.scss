.testimonials {
    margin-top: 64px;
    margin-left: 64px;
    margin-right: 64px;
    background-color: #FCFCFC;
}

.list {
    padding-left: 64px;
    padding-right: 64px;
}
