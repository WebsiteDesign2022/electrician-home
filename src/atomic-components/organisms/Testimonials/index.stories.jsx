import React from "react";

import Testimonials from ".";

export default {
    title: 'Electrician-Home/organisms/Testimonials',
    component: Testimonials,
    parameters: {
      // More on Story layout: https://storybook.js.org/docs/react/configure/story-layout
      layout: 'fullscreen',
    },
};

const Template = (args) => <Testimonials {...args} />

Template.args = {
};

export const Primary = Template.bind({});
Primary.args = {
    testimonials: [
        {
            name: 'Ivan',
            description: 'Testimonial description from Ivan'
        },
        {
            name: 'Shelley',
            description: 'Testimonial description from Shelley'
        }
    ]
};
