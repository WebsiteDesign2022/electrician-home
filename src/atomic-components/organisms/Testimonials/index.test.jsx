import TestRenderer from 'react-test-renderer';

import Testimonials from '.';

describe("Testimonials", () => {
    test("renders Testimonials given empty state", () => {
        const render = TestRenderer.create(
            <Testimonials
                testimonials={[]}
            />
        );
        expect(render.toJSON()).toMatchSnapshot();
    });

    test("renders Testimonials given non-empty state", () => {
        const render = TestRenderer.create(
            <Testimonials
                testimonials={[
                    {
                        name: 'testimonial name 1',
                        description: 'testimonial description 1'
                    }
                ]}
            />
        );
        expect(render.toJSON()).toMatchSnapshot();
    });

    test("renders Testimonials given non-empty state", () => {
        const render = TestRenderer.create(
            <Testimonials
                testimonials={[
                    {
                        name: 'testimonial name 1',
                        description: 'testimonial description 1'
                    },
                    {
                        name: 'testimonial name 2',
                        description: 'testimonial description 2'
                    },
                    {
                        name: 'testimonial name 3',
                        description: 'testimonial description 3'
                    },
                    {
                        name: 'testimonial name 4',
                        description: 'testimonial description 4'
                    }
                ]}
            />
        );
        expect(render.toJSON()).toMatchSnapshot();
    });
});
