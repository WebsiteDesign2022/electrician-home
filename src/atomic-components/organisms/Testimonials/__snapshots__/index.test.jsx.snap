// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testimonials renders Testimonials given empty state 1`] = `
<div
  className="testimonials"
>
  <div
    className="border-bottom align-content-start border-dark align-content-start col-3"
  >
    <div
      className="arrow-top-left-black"
    />
    <h3
      align="left"
    >
      Testimonials
    </h3>
  </div>
  <div
    className="row row-cols-1 row-cols-md-3 gx-4 py-5 list"
  />
</div>
`;

exports[`Testimonials renders Testimonials given non-empty state 1`] = `
<div
  className="testimonials"
>
  <div
    className="border-bottom align-content-start border-dark align-content-start col-3"
  >
    <div
      className="arrow-top-left-black"
    />
    <h3
      align="left"
    >
      Testimonials
    </h3>
  </div>
  <div
    className="row row-cols-1 row-cols-md-3 gx-4 py-5 list"
  >
    <div
      className="col d-flex align-items-start px-6 gx-6 bg-white testimonial"
    >
      <div
        className="m-12"
      >
        <p
          align="left"
          className="testimonial-content"
        >
          testimonial description 1
        </p>
        <h6
          align="left"
        >
          testimonial name 1
        </h6>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testimonials renders Testimonials given non-empty state 2`] = `
<div
  className="testimonials"
>
  <div
    className="border-bottom align-content-start border-dark align-content-start col-3"
  >
    <div
      className="arrow-top-left-black"
    />
    <h3
      align="left"
    >
      Testimonials
    </h3>
  </div>
  <div
    className="row row-cols-1 row-cols-md-3 gx-4 py-5 list"
  >
    <div
      className="col d-flex align-items-start px-6 gx-6 bg-white testimonial"
    >
      <div
        className="m-12"
      >
        <p
          align="left"
          className="testimonial-content"
        >
          testimonial description 1
        </p>
        <h6
          align="left"
        >
          testimonial name 1
        </h6>
      </div>
    </div>
    <div
      className="col d-flex align-items-start px-6 gx-6 bg-white testimonial"
    >
      <div
        className="m-12"
      >
        <p
          align="left"
          className="testimonial-content"
        >
          testimonial description 2
        </p>
        <h6
          align="left"
        >
          testimonial name 2
        </h6>
      </div>
    </div>
    <div
      className="col d-flex align-items-start px-6 gx-6 bg-white testimonial"
    >
      <div
        className="m-12"
      >
        <p
          align="left"
          className="testimonial-content"
        >
          testimonial description 3
        </p>
        <h6
          align="left"
        >
          testimonial name 3
        </h6>
      </div>
    </div>
    <div
      className="col d-flex align-items-start px-6 gx-6 bg-white testimonial"
    >
      <div
        className="m-12"
      >
        <p
          align="left"
          className="testimonial-content"
        >
          testimonial description 4
        </p>
        <h6
          align="left"
        >
          testimonial name 4
        </h6>
      </div>
    </div>
  </div>
</div>
`;
